### DISCUSSION TITLE
ESP32 resets to default settings despite setoption65 to 1

### DISCUSSION THREAD

**Discussion by Bluescreen2001** [arendst/Tasmota#17504] [Q&A] [answered]
Hi
I want to play a bit with Tasmota and KNX. So I took three esp32 modules and flashed them via the online flash utility with the KNX version of Tasmota.
I can configure the device completely and it then works for a few seconds to minutes. But then it resets and loose all it's settings. I read something of the auto recovery and for safety disabled that function by issueing setoption 65 1. But with no success. I didn't connect any hardwar to the device yet. Only use the devkit module connectet via USB to my PC.

All three modules show the same issue.
Do you have a idea what to do?

I add a log below with a complete configuration process from connecting to WIFI until crash. I commented it a bit. Search for ### to find my comments.

ets Jul 29 2019 12:21:46

rst:0x1 (POWERON_RESET),boot:0x1f (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:184
load:0x40078000,len:12588
ho 0 tail 12 room 4
load:0x40080400,len:2916
entry 0x400805c4

00:00:00.002 HDW: ESP32-D0WD-V3
00:00:00.036 UFS: FlashFS mounted with 304 kB free
00:00:00.065 CFG: Loaded from File, Anzahl 5
00:00:00.075 QPC: Count 1
00:00:00.172 BRY: Berry initialized, RAM used=4041 bytes
00:00:00.191 Projekt tasmota - Tasmota Version 12.3.1(tasmota)-2_0_5_3(2022.12.1                                                                                  6 10:36:41)
00:00:00.514 WIF: WifiManager aktiv für 3 Minuten
00:00:01.197 HTP: Web-Server aktiv bei tasmota-FD8078-0120 mit IP-Adresse 192.16                                                                                  8.4.1
00:00:06.952 QPC: Reset
00:01:08.144 WIF: verbinden mit AP lcars als tasmota-FD8078-0120 ...
00:01:10.613 WIF: SSId lcars: verbunden - IP-Adresse 192.168.42.142
00:01:22.061 APP: starte neu
ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x1f (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:184
load:0x40078000,len:12588
ho 0 tail 12 room 4
load:0x40080400,len:2916
entry 0x400805c4

00:00:00.002 HDW: ESP32-D0WD-V3
00:00:00.052 UFS: FlashFS mounted with 304 kB free
00:00:00.103 CFG: Loaded from File, Anzahl 7
00:00:00.113 QPC: Count 1
00:00:00.242 BRY: Berry initialized, RAM used=4041 bytes
00:00:00.263 Projekt tasmota - Tasmota Version 12.3.1(tasmota)-2_0_5_3(2022.12.16 10:36:41)
00:00:00.974 WIF: verbinden mit AP1 lcars in Modus 11n als tasmota-FD8078-0120...
00:00:02.123 WIF: verbunden
00:00:02.384 HTP: Web-Server aktiv bei tasmota-FD8078-0120 mit IP-Adresse 192.168.42.142
11:51:32.900 RSL: INFO1 = {"Info1":{"Module":"ESP32-DevKit","Version":"12.3.1(tasmota)","FallbackTopic":"cmnd/DVES_FD8078_fb/","GroupTopic":"cmnd/tasmotas/"}}
11:51:32.913 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-FD8078-0120","IPAddress":"192.168.42.142"}}
11:51:32.926 RSL: INFO3 = {"Info3":{"RestartReason":"Software reset CPU","BootCount":4}}
11:51:35.909 QPC: Reset
11:51:36.906 RSL: STATE = {"Time":"2022-12-25T11:51:36","Uptime":"0T00:00:08","UptimeSec":8,"Heap":153,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":43},"Wifi":{"AP":1,"SSId":"lcars","BSSId":"C8:0E:14:58:6B:5E","Channel":9,"Mode":"11n","RSSI":100,"Signal":-27,"LinkCount":1,"Downtime":"0T00:00:03"}}
11:52:53.305 RSL: RESULT = {"Hostname":"KNX1"}  ### change Hostname
11:52:53.317 RSL: RESULT = {"SSId1":"lcars"}
11:52:53.323 RSL: RESULT = {"SSId2":""}
11:52:53.329 RSL: RESULT = {"Password1":"****"}
11:52:53.335 RSL: RESULT = {"Password2":"****"}
11:52:54.684 APP: starte neu
ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x1f (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:184
load:0x40078000,len:12588
ho 0 tail 12 room 4
load:0x40080400,len:2916
entry 0x400805c4

00:00:00.002 HDW: ESP32-D0WD-V3
00:00:00.020 UFS: FlashFS mounted with 304 kB free
00:00:00.044 CFG: Loaded from File, Anzahl 10
00:00:00.054 QPC: Count 1
00:00:00.131 BRY: Berry initialized, RAM used=4041 bytes
00:00:00.149 Projekt tasmota - Tasmota Version 12.3.1(tasmota)-2_0_5_3(2022.12.16 10:36:41)
00:00:00.693 WIF: verbinden mit AP1 lcars Channel 9 BSSId C8:0E:14:58:6B:5E in Modus 11n als KNX1...
00:00:02.438 WIF: verbunden
00:00:02.695 HTP: Web-Server aktiv bei KNX1 mit IP-Adresse 192.168.42.142
11:52:59.710 RSL: INFO1 = {"Info1":{"Module":"ESP32-DevKit","Version":"12.3.1(tasmota)","FallbackTopic":"cmnd/DVES_FD8078_fb/","GroupTopic":"cmnd/tasmotas/"}}
11:52:59.724 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"KNX1","IPAddress":"192.168.42.142"}}
11:52:59.726 RSL: INFO3 = {"Info3":{"RestartReason":"Software reset CPU","BootCount":5}}
11:53:02.974 QPC: Reset
11:53:03.971 RSL: STATE = {"Time":"2022-12-25T11:53:03","Uptime":"0T00:00:08","UptimeSec":8,"Heap":153,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":43},"Wifi":{"AP":1,"SSId":"lcars","BSSId":"C8:0E:14:58:6B:5E","Channel":9,"Mode":"11n","RSSI":100,"Signal":-28,"LinkCount":1,"Downtime":"0T00:00:03"}}
11:53:57.276 RSL: RESULT = {"WebPassword":"****"}
11:53:57.381 RSL: RESULT = {"SetOption3":"ON"}
11:53:57.392 RSL: RESULT = {"SetOption128":"ON"}
11:53:57.398 RSL: RESULT = {"DeviceName":"KNX1"}  ### change Devicename
11:53:57.405 RSL: RESULT = {"FriendlyName1":"KNX Taster links"}   ### change Friendlyname
11:53:57.411 RSL: RESULT = {"FriendlyName2":""}
11:53:57.417 RSL: RESULT = {"FriendlyName3":""}
11:53:57.423 RSL: RESULT = {"FriendlyName4":""}
11:53:57.432 RSL: RESULT = {"FriendlyName5":""}
11:53:57.442 RSL: RESULT = {"FriendlyName6":""}
11:53:57.448 RSL: RESULT = {"FriendlyName7":""}
11:53:57.454 RSL: RESULT = {"FriendlyName8":""}
11:53:57.460 RSL: RESULT = {"Emulation":0}
11:53:57.469 RSL: RESULT = {"NAME":"ESP32-DevKit","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,1,1,1,0,1,1,1,0,0,0,0,1,1,1,1,1,0,0,1],"FLAG":0,"BASE":1}
11:53:59.188 APP: starte neu
ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x1f (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:184
load:0x40078000,len:12588
ho 0 tail 12 room 4
load:0x40080400,len:2916
entry 0x400805c4

00:00:00.002 HDW: ESP32-D0WD-V3
00:00:00.046 UFS: FlashFS mounted with 304 kB free
00:00:00.086 CFG: Loaded from File, Anzahl 13
00:00:00.096 QPC: Count 1
00:00:00.213 BRY: Berry initialized, RAM used=4041 bytes
00:00:00.234 Projekt tasmota - KNX1 Version 12.3.1(tasmota)-2_0_5_3(2022.12.16 10:36:41)
00:00:00.931 WIF: verbinden mit AP1 lcars Channel 9 BSSId C8:0E:14:58:6B:5E in Modus 11n als KNX1...
00:00:01.139 WIF: verbunden
00:00:01.396 HTP: Web-Server aktiv bei KNX1 mit IP-Adresse 192.168.42.142
11:54:01.907 RSL: INFO1 = {"Info1":{"Module":"ESP32-DevKit","Version":"12.3.1(tasmota)","FallbackTopic":"cmnd/DVES_FD8078_fb/","GroupTopic":"cmnd/tasmotas/"}}
11:54:01.921 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"KNX1","IPAddress":"192.168.42.142"}}
11:54:01.924 RSL: INFO3 = {"Info3":{"RestartReason":"Software reset CPU","BootCount":6}}
11:54:05.917 QPC: Reset
11:54:05.924 RSL: STATE = {"Time":"2022-12-25T11:54:05","Uptime":"0T00:00:07","UptimeSec":7,"Heap":153,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":43},"Wifi":{"AP":1,"SSId":"lcars","BSSId":"C8:0E:14:58:6B:5E","Channel":9,"Mode":"11n","RSSI":100,"Signal":-28,"LinkCount":1,"Downtime":"0T00:00:02"}}
11:54:27.550 CMD: Setoption65 1  ### switch off Brownout aut recovery
11:54:27.553 RSL: RESULT = {"SetOption65":"ON"}
11:54:43.120 CMD: Save config
11:54:43.127 RSL: RESULT = {"Command":"Unknown"}
11:54:50.223 CMD: Save
11:54:50.230 RSL: RESULT = {"Command":"Unknown"}
11:56:47.142 RSL: RESULT = {"NAME":"KNX Taster","GPIO":[256,1,257,1,258,259,1,1,96,97,98,99,1,1,1,1,0,1,1,1,0,1,1,1,0,0,0,0,1,1,1,1,1,0,0,1],"FLAG":0,"BASE":1}  ### create profile "KNX Taster"
11:56:48.635 APP: starte neu
ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x1f (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:184
load:0x40078000,len:12588
ho 0 tail 12 room 4
load:0x40080400,len:2916
entry 0x400805c4

00:00:00.002 HDW: ESP32-D0WD-V3
00:00:00.016 UFS: FlashFS mounted with 304 kB free
00:00:00.034 CFG: Loaded from File, Anzahl 16
00:00:00.100 BRY: Berry initialized, RAM used=4041 bytes
00:00:00.116 Projekt tasmota - KNX1 Version 12.3.1(tasmota)-2_0_5_3(2022.12.16 10:36:41)
00:00:00.719 WIF: verbinden mit AP1 lcars Channel 9 BSSId C8:0E:14:58:6B:5E in Modus 11n als KNX1...
00:00:01.086 WIF: verbunden
00:00:01.347 HTP: Web-Server aktiv bei KNX1 mit IP-Adresse 192.168.42.142
11:56:52.712 RSL: INFO1 = {"Info1":{"Module":"ESP32-DevKit","Version":"12.3.1(tasmota)","FallbackTopic":"cmnd/DVES_FD8078_fb/","GroupTopic":"cmnd/tasmotas/"}}
11:56:52.726 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"KNX1","IPAddress":"192.168.42.142"}}
11:56:52.728 RSL: INFO3 = {"Info3":{"RestartReason":"Software reset CPU","BootCount":7}}
11:56:55.951 RSL: STATE = {"Time":"2022-12-25T11:56:55","Uptime":"0T00:00:06","UptimeSec":6,"Heap":153,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":43},"Wifi":{"AP":1,"SSId":"lcars","BSSId":"C8:0E:14:58:6B:5E","Channel":9,"Mode":"11n","RSSI":100,"Signal":-26,"LinkCount":1,"Downtime":"0T00:00:02"}}
11:57:22.614 RSL: RESULT = {"Module":{"0":"KNX Taster"}}  ### set device to profil "KNX Taster"
11:57:22.630 RSL: RESULT = {"GPIO1":{"0":"None"},"GPIO3":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"},"GPIO18":{"0":"None"},"GPIO19":{"0":"None"},"GPIO21":{"0":"None"},"GPIO22":{"0":"None"},"GPIO23":{"0":"None"},"GPIO25":{"0":"None"},"GPIO26":{"0":"None"},"GPIO27":{"0":"None"},"GPIO32":{"0":"None"},"GPIO33":{"0":"None"},"GPIO34":{"0":"None"},"GPIO35":{"0":"None"},"GPIO36":{"0":"None"},"GPIO39":{"0":"None"}}
11:57:24.207 APP: starte neu
ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x1f (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:184
load:0x40078000,len:12588
ho 0 tail 12 room 4
load:0x40080400,len:2916
entry 0x400805c4

00:00:00.003 HDW: ESP32-D0WD-V3
00:00:00.031 UFS: FlashFS mounted with 304 kB free
00:00:00.060 CFG: Loaded from File, Anzahl 18
00:00:00.152 BRY: Berry initialized, RAM used=4041 bytes
00:00:00.173 Projekt tasmota - KNX1 Version 12.3.1(tasmota)-2_0_5_3(2022.12.16 10:36:41)
00:00:00.943 WIF: verbinden mit AP1 lcars Channel 9 BSSId C8:0E:14:58:6B:5E in Modus 11n als KNX1...
00:00:01.090 WIF: verbunden
00:00:01.351 HTP: Web-Server aktiv bei KNX1 mit IP-Adresse 192.168.42.142
11:57:27.961 RSL: INFO1 = {"Info1":{"Module":"KNX Taster","Version":"12.3.1(tasmota)","FallbackTopic":"cmnd/DVES_FD8078_fb/","GroupTopic":"cmnd/tasmotas/"}}
11:57:27.975 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"KNX1","IPAddress":"192.168.42.142"}}
11:57:27.977 RSL: INFO3 = {"Info3":{"RestartReason":"Software reset CPU","BootCount":8}}
11:57:27.990 RSL: RESULT = {"POWER1":"OFF"}
11:57:27.991 RSL: POWER1 = OFF
11:57:27.992 RSL: RESULT = {"POWER2":"OFF"}
11:57:28.003 RSL: POWER2 = OFF
11:57:28.004 RSL: RESULT = {"POWER3":"OFF"}
11:57:28.005 RSL: POWER3 = OFF
11:57:28.006 RSL: RESULT = {"POWER4":"OFF"}
11:57:28.007 RSL: POWER4 = OFF
11:57:31.949 RSL: STATE = {"Time":"2022-12-25T11:57:31","Uptime":"0T00:00:07","UptimeSec":7,"Heap":153,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":26,"MqttCount":0,"Berry":{"HeapUsed":3,"Objects":43},"POWER1":"OFF","POWER2":"OFF","POWER3":"OFF","POWER4":"OFF","Wifi":{"AP":1,"SSId":"lcars","BSSId":"C8:0E:14:58:6B:5E","Channel":9,"Mode":"11n","RSSI":100,"Signal":-28,"LinkCount":1,"Downtime":"0T00:00:02"}}
### everything works now. I pressed some buttons in the web UI and the pins change accordingly
11:57:51.607 RSL: RESULT = {"POWER1":"ON"}
11:57:51.609 RSL: POWER1 = ON
11:57:55.059 RSL: RESULT = {"POWER1":"OFF"}
11:57:55.061 RSL: POWER1 = OFF
11:57:56.413 RSL: RESULT = {"POWER2":"ON"}
11:57:56.414 RSL: POWER2 = ON
11:57:58.957 RSL: RESULT = {"POWER2":"OFF"}
11:57:58.959 RSL: POWER2 = OFF
11:57:59.975 RSL: RESULT = {"POWER3":"ON"}
11:57:59.976 RSL: POWER3 = ON
11:58:01.634 RSL: RESULT = {"POWER3":"OFF"}
11:58:01.636 RSL: POWER3 = OFF
11:58:02.847 RSL: RESULT = {"POWER4":"ON"}
11:58:02.848 RSL: POWER4 = ON
11:58:04.415 RSL: RESULT = {"POWER4":"OFF"}
11:58:04.416 RSL: POWER4 = OFF
11:58:10.495 RSL: RESULT = {"Reset":"Reset and Restarting"}
11:58:10.501 RSL: RESULT = {"Reset":"Reset and Restarting"}
11:58:10.506 RSL: RESULT = {"Reset":"Reset and Restarting"}
11:58:10.512 RSL: RESULT = {"Reset":"Reset and Restarting"}
11:58:11.451 CFG: Standard verwenden
11:58:13.449 APP: starte neu
ets Jul 29 2019 12:21:46

### crash after this everything is gone ###

rst:0xc (SW_CPU_RESET),boot:0x1f (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:184
load:0x40078000,len:12588
ho 0 tail 12 room 4
load:0x40080400,len:2916
entry 0x400805c4

00:00:00.002 HDW: ESP32-D0WD-V3
00:00:00.037 UFS: FlashFS mounted with 304 kB free
00:00:00.072 CFG: Loaded from File, Anzahl 2
00:00:00.082 QPC: Count 1
00:00:00.468 BRY: Berry initialized, RAM used=4711 bytes
00:00:00.489 Projekt tasmota - Tasmota Version 12.3.1(tasmota)-2_0_5_3(2022.12.16 10:36:41)
00:00:00.014 WIF: WifiManager aktiv für 3 Minuten
00:00:01.689 HTP: Web-Server aktiv bei tasmota-FD8078-0120 mit IP-Adresse 192.168.4.1
00:00:06.948 QPC: Reset

**Comment by barbudor** [comment]
For me everything looks good until
11:58:10.495 RSL: RESULT = {"Reset":"Reset and Restarting"}

All other reboots seems to follow application of setting or command that requires reboot

What exact template have you applied?

**Reply by Bluescreen2001** [reply]
Yes, you are right. All the Resets are ok until the last one, which deletes everything.

I created a profile named "KNX Taster" with the following settings:

GPIO 0, 2, 4, 5 : 4x RELAIS_i
GPIO 12, 13, 14, 15 : 4x BUTTON_i

I also tried to exchange the relais and buttons. I didn't connect anything yet to the device's GPIOs.
Another thing I tried is to do the GPIO settings directly from the default profile ESP32-DevKit, without using a new profile. But with the same issue.

**Reply by barbudor** [reply]
By Button_i do you mean 'i' is the index 1 to 4 or 'inverted'
If inverted, You just falled into the 40 seconds button press recovery
As the GPIOs are by default with a pull up, Button_i 1 is seen pressed and trigger a reset after 40+ seconds

Can only be disabled by setoption1 3 which also disable multi press

**Comment by Bluescreen2001** [comment]
I mean inverted. I thought I had to invert if the buttons are connected to 
ground. So, you mean if I connect buttons to GND as I do i have to NOT 
invert them? I am just not at homeand will try in 30 minutes or so. Btw.: 
What donI have to use if the buttons should switch the relais?

Am 25. Dezember 2022 14:12:20 schrieb Barbudor ***@***.***>:

**Reply by barbudor** [reply]
Normal buttons, not inverted are between gpio and gnd (with internal pull up). So active low
Inverted are active high.
That might sound counter intuitive but most generally buttons are connected this way.

If your buttons are set as not inverted, and connected to gnd, then you don't need so13. Settings reset would occur only if pressing more than 40 secs

On the other side Relay is active high and Relay_i is active low 🤣

By default Button N drives Relay N
If you don't want that you can decouple buttons from relay. There are few options for that in https://tasmota.github.io/docs/Buttons-and-Switches/

🎅

**Comment by Bluescreen2001** [comment]
I now try with four times BUTTON_N without inverting. But it crashes again. Then I tried only with one BUTTON_N on GPIO0 and one RELAIS_I on GPIO12. It is now stable for a few minutes. I can switch the Relais via Web but not via button. Both channels, BUTTON_N and RELAIS_I are set to one.

**Reply by barbudor** [reply]
Why Button_N?
Why not just Button???
No pull up on an unconnected gpio gets random

**Comment by Bluescreen2001** [comment]
Hm. I should read the docs in more detail. I thought BUTTON_I is inverted and BUTTON_N is non inverted but connected to relais. And only BUTTON is not connected to any relais. Can it be a problem to use the lower GPIOs for output like relais because some of them are used for bootstrapping the ESP?

**Reply by barbudor** [reply]
Button drives the relay independently of hardware option. Decoupling buttons from relay is not done in the GPIO configuration

Button (as in momentary push button) can be connected to those gpios (between gpio and gnd) as far as no one press the button during boot

Never connected a flip-switch (aka rocker switch) to such gpio.

Yes please read carefully the buttons and switch page. 3 times better than 1 😂

**Comment by Bluescreen2001** [comment]
I now changed from BUTTON_N to BUTTON and can now switch the Relais without any reset now. I will now add the buttons one by one to check if ther is an issue with one special GPIO.

**Comment by Bluescreen2001** [comment, marked as answer]
Now it seems to work. Thanks very much for help.
May I ask another question:

If I define some PWM outputs, I get some color setting controls. But I want to have simple brightness sliders, not color. Is that possible?

Is it possible to hide the menu on the web interface and only have the button or slider controls? So a normal user is not confronted with all the settings.

**Reply by barbudor** [reply]
Pwm: there's a SetOption for that, search SetOption68

UI: there a user mode (vs default admin mode) that can be enabled with command WebServer

**Reply by sfromis** [reply]
Beware that if you enable user mode, you not change back to admin mode using the web UI. For that you'll need to issue the command via MQTT or serial, or possibly have some rule setup to switch it depending on some condition, maybe another button.

**Reply by Bluescreen2001** [reply]
Good hints. Thank you.
I now have four single brightness sliders.
Didn't try User UI now, but at the moment I can use serial console for switching back. Later this is not possible. I want to use it in a MQTT environment, so it should be possible to reset it from there. Cool would be to have a button like a cogwheel or something or even an invisible button. Pressing that brings up a password dialog and then enters the config menus. But we cannot have everthing we like. :-)

I have some more questions, but I have to first lear a bit more now. Maybe I find the answers in the docs. So thank you for help for now.

Thorsten

**Reply by sfromis** [reply]
Tasmota is not designed for the web UI being for more than admin/test, not for "daily operation" in an user-friendly manner
